src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -55,27 +56,40 @@ def __init__(
         self._migration_status_refresher = migration_status_refresher
         self._seen_tables: dict[str, str] = {}
         self._principal_grants = principal_grants
+        self._hiveserde_in_place_migrate = hiveserde_in_place_migrate
```

Comment:
```suggestion
```

I'm still not convinced that we need another instance of `TableMigrator` with this property being different. 

make `hiveserde_in_place_migrate` as an argument to `migrate_tables`. it'll be easier to replace later with with a strategy pattern.

Reply:
while it was the case when this PR was originally created, the `hiveserde_in_place_migrate` was the an argument to `migrate_tables`. But you commented to put it to the constructor, because it has to be passed over from `migrate_tables` to `_migrate_tables` to `_migrate_table` to `_migrate_external_table_hiveserde` :joy:
Anyway, since know we all agree it's better to have it as the `migrate_tables` function argument, I will refactor it.

Reply:
I think the comment was about MountsCrawler